### ISSUE TITLE
Can't seem to get this to work

### ISSUE THREAD

**Issue by skielbasa** [shmargum/vim-sass-colors#6]
After installing with minpac on vim8, nothing happens when opening scss files. I thought it might be something with my folder structure, so I tested the file that has all the definitions it and those wouldn't highlight either.

If I run `:echo has('ruby')` I get `1` and when running `vim --version | grep 'ruby'` I get `+ruby` as part of the output. That should mean my vim has ruby support, correct?

Running `:set filetype?` on any of the files I've tested outputs `filetype=scss`, so I don't think there's anything wrong with miss-identification.

I have truecolors enabled in Vim and I am using a truecolor compatible terminal.

I've verified that the plugin's directory is in my run time path and I have `filetype plugin indent on` in my .vimrc file. I'm pretty new to Vim, so that's about all I've got for trying to figure this out.

* Vim version: VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Oct 17 2017 11:21:47) Included Patches: 1-1300 

* Ruby version: ruby 2.4.2p198 (2017-09-14 revision 59899) [x86_64-darwin16]

* OS version: MacOS 10.13.1 (17B48)

**Comment by shmargum** [comment]
first thing I am wondering is if the plugin ran
try running `:echo b:did_vim_sass_colors`
If the plugin was loaded the output should be 1, otherwise you should get an undefined error

if the output is undefined, then the issue is with loading the plugin, otherwise the issue lies within the plugin

**Comment by skielbasa** [comment]
> try running :echo b:did_vim_sass_colors from within a .scss file

It returns a `1`

> you may also need a syntax on line in your .vimrc if you don't have it

I've got `syntax enable` in my .vimrc. I assume that does the same thing, I've got functioning syntax highlighting.

If it helps, I get the follow error when saving .scss files with the plugin running 
```
Error detected while processing function vim_sass_colors#write:
line    2:
E803: ID not found: 81
E803: ID not found: 82
E803: ID not found: 83
E803: ID not found: 84
E803: ID not found: 85
E803: ID not found: 86
E803: ID not found: 87
E803: ID not found: 88
E803: ID not found: 89
E803: ID not found: 90
E803: ID not found: 91
E803: ID not found: 92
E803: ID not found: 93
E803: ID not found: 94
E803: ID not found: 95
E803: ID not found: 96
E803: ID not found: 97
E803: ID not found: 98
E803: ID not found: 99
```

**Comment by shmargum** [comment]
I've pushed a new version up, it resolves an issue when a file is found matching the import name but with a wrong suffix.  Not sure if this is the issue you were experiencing, but it could have caused this error.

The errors indicate to me that the matches do indeed get loaded, and added to an array, but not added to the actual highlight group.

I have not seen these particular errors, so it would help if you add some echo statements in the vim file to see where things are going awry, somewhere in here:
https://github.com/shmargum/vim-sass-colors/blob/master/autoload/vim_sass_colors.vim#L34-L46

**Comment by shmargum** [comment]
When you open your file and run `:hi` it will print the highlight groups.
If you scroll down to the bottom, the highlight groups added should start with SASS (capitalized) and then have color value after like 000000 (for black)
Do you see this, or are there no such values?

**Comment by shmargum** [comment]
OK i seem to have reproduced this issue.
It appears when I open a file, then do a vim split `:vs` and hit enter to edit the same file, then I try to save the new buffer.

Will look into this more

**Comment by skielbasa** [comment]
> ...if you add some echo statements...

Sorry, I'm not completely sure what I should be echoing.

> When you open your file and run `:hi` it will print the highlight groups.

Running this, I saw the following towards the beginning of its ouptut:

```
SASSf1f1f1     xxx ctermfg=16 ctermbg=255 guifg=#000000 guibg=#f1f1f1
SASSe6e6e6     xxx ctermfg=16 ctermbg=254 guifg=#000000 guibg=#e6e6e6
SASSfff9c0     xxx ctermfg=16 ctermbg=229 guifg=#000000 guibg=#fff9c0
SASS21759b     xxx ctermfg=15 ctermbg=30 guifg=#ffffff guibg=#21759b
SASS404040     xxx ctermfg=15 ctermbg=238 guifg=#ffffff guibg=#404040
SASS00a5e4     xxx ctermfg=16 ctermbg=38 guifg=#000000 guibg=#00a5e4
SASS096796     xxx ctermfg=15 ctermbg=24 guifg=#ffffff guibg=#096796
SASS4e88ff     xxx ctermfg=16 ctermbg=69 guifg=#000000 guibg=#4e88ff
SASS3e41ff     xxx ctermfg=15 ctermbg=63 guifg=#ffffff guibg=#3e41ff
SASS2a2a2d     xxx ctermfg=15 ctermbg=235 guifg=#ffffff guibg=#2a2a2d
SASS838d8d     xxx ctermfg=16 ctermbg=245 guifg=#000000 guibg=#838d8d
SASSebebeb     xxx ctermfg=16 ctermbg=255 guifg=#000000 guibg=#ebebeb
SASSfefefe     xxx ctermfg=16 ctermbg=231 guifg=#000000 guibg=#fefefe
SASSf7f9fc     xxx ctermfg=16 ctermbg=231 guifg=#000000 guibg=#f7f9fc
SASS6a8da1     xxx ctermfg=16 ctermbg=67 guifg=#000000 guibg=#6a8da1
SASSff4b00     xxx ctermfg=16 ctermbg=202 guifg=#000000 guibg=#ff4b00
SASS5fbaab     xxx ctermfg=16 ctermbg=73 guifg=#000000 guibg=#5fbaab
SASS6fe793     xxx ctermfg=16 ctermbg=78 guifg=#000000 guibg=#6fe793
SASSecbd1c     xxx ctermfg=16 ctermbg=214 guifg=#000000 guibg=#ecbd1c
```

When I was looking through the output, I noticed some similar highlight groups for CSS. Thinking maybe it was a conflict, I uninstalled Coloresque and highlighting for SASS variables started working. 

Sorry if I've been wasting your time on this. I figured they were compatible (the issues of the Coloresque is where I actually found your plugin). Do they just not play nicely or did I set something up wrong?

As some background, [pigments](https://github.com/abe33/atom-pigments) is one of the last plugins/features from Atom that I'm trying to get in Vim. To me, your plugin for SCSS + Coloresque for pretty much everything else seemed to fill that void.

> OK i seem to have reproduced this issue.
It appears when I open a file, then do a vim split :vs and hit enter to edit the same file, then I try to save the new buffer. Will look into this more

Amusingly, you sent that right before I hit submit on this message. As for splits, for this actual testing, I haven't been using any. While that may be an issue since you were able to trigger that, I thinking it was a compatibility thing in my case.

Also, I just wanted to add that I appreciate you taking the time to try and help me solve this.

**Comment by shmargum** [comment]
Thanks for pointing these findings out, it's very helpful.
I think I can address all these issues.
I also intend to add the rest of the color highlighting features and make this work for regular css files, because why not :)

**Comment by skielbasa** [comment]
> Thanks for pointing these findings out, it's very helpful.

If there's anything else I can do to help, let me know. Unfortunately, I don't know enough ruby or vimscript to be of much help though.

> I think I can address all these issues.
I also intend to add the rest of the color highlighting features and make this work for regular css files, because why not :)

Again, thank you. Very much appreciated!

**Comment by shmargum** [comment]
I see why this would not work with vim coloresque, they do `call clearmatches()`
on initialization which removes all highlight groups.

If that plugin was loaded first, then I think the two would be compatible.

I think this issue can be raised with that plugin, since there is not really much I can do to prevent another plugin from clobbering all the highlight groups

I have added a few commits though that do resolve the issue I described previously about using vim split.

Gonna close this issue, since the problem appears to be in the way vim coloresque clears their matches.  In this plugin we only clear matches that were added by itself, vs clearing all matches which is what coloresque does.  Additionally, overall the colorsque autocmd hooks seem somewhat inefficient.
